sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -396,6 +396,13 @@ object SQLConf {
       .intConf
       .createWithDefault(32)
 
+  val PARALLEL_PARTITION_DISCOVERY_PARALLELISM =
+    SQLConfigBuilder("spark.sql.sources.parallelPartitionDiscovery.parallelism")
+      .doc("The number of parallelism to list a collection of path recursively, Set the " +
+        "number to prevent file listing from generating too many tasks.")
+      .intConf
+      .createWithDefault(100)
```

Comment:
IMHO, 10000 is too large for the file discovery in consideration of scene where cluster is not very large, and it makes	 no contributions to improve performance in most scenes.